File: server/api/user_data.py
from models.job import Job
from models.section import Section
from models.study import Study
from typing import List

class UserDataResponse:
    def __init__(self, jobs: List[Job], studies: List[Study], otherSections: List[Section], skills: List[str]) -> None:
        self.jobs = jobs
        self.studies = studies
        self.otherSections = otherSections
        self.skills = skills

def get_user_data_response(userData: UserDataResponse) -> dict:
    jobs = [job.toDict() for job in userData.jobs]
    studies = [study.toDict() for study in userData.studies]
    otherSections = [section.toDict() for section in userData.otherSections]
    return {
        'jobs': jobs,
        'studies': studies,
        'skills': userData.skills,
        'otherSections': otherSections
    }

File: client_final/src/models/userData.ts
import dayjs from "dayjs";

export type Job = {
  company: string;
  position: string;
  description: string;
  startDate: string;
  endDate: string;
};

export type Section = {
  title: string;
  content: string;
};

export type Study = {
  institution: string;
  degree: string;
  startDate: string;
  endDate: string;
};

export type UserData = {
  _id: string;
  jobs: Job[];
  studies: Study[];
  skills: string[];
  otherSections: Section[];
};

export const CarrerModelProperties = [
  "company",
  "position",
  "description",
  "startDate",
  "endDate",
];

export const SectionModelProperties = ["title", "content"];

export const StudyModelProperties = [
  "institution",
  "degree",
  "startDate",
  "endDate",
];

export enum UserDataFormSections {
  Studies = "STUDIES",
  Jobs = "JOBS",
  Others = "OTHER_SECTIONS",
  Skills = "SKILLS",
}

export type UserFormData = Record<string, string | dayjs.Dayjs>;
